### ISSUE TITLE
SCEdit not converge after 2w iterations on COCO dataset.

### ISSUE THREAD

**Issue by serend1p1ty** [modelscope/scepter#26]
I have trained SCEdit [2w/5w, 40%] iterations on COCO dataset.

I observed the loss is always around 0.4. The model have not demonstrated controllable capabilities.

![image](https://github.com/modelscope/scepter/assets/39628662/904deb0c-5e27-466f-a20c-9fbebcbbdc1f)

![image](https://github.com/modelscope/scepter/assets/39628662/2dd2dd10-67ac-46ee-a872-fd858aa2d4a4)

Is it normal?

My configuration (sd21_768_sce_ctr_canny.yaml, 2 GPUs):

```
max_steps=50000
batch_size=4
lr=1e-5
```

**Comment by jiangzeyinzi** [comment]
Sorry for the inconvenience this may have caused you. The problem could be due to the learning rate being set too low, which might be affecting convergence. Our framework employs a batch learning rate to accommodate multi-GPU scenarios, which is calculated as "real_lr = yaml_lr * gpu_num * batch / 640". This can also be observed in the log entries under "pg0_lr: 0.xxx". We plan to make adjustments to ensure compatibility with this learning rate setting in the future. 
For more information, please see: https://github.com/modelscope/scepter/blob/main/scepter/modules/solver/diffusion_solver.py#L136

**Comment by serend1p1ty** [comment]
@jiangzeyinzi Thanks for your reply. Can you tell me approximately what the final loss value is.
Current value is 0.37. Is it normal?

![image](https://github.com/modelscope/scepter/assets/39628662/fdab4239-669f-44a0-8ad0-af258bbe14af)

**Comment by jiangzeyinzi** [comment]
In generative tasks, loss often cannot serve as the central basis for model convergence. In our setup, the loss went from 0.16 to 0.14. Additionally, the loss also shows apparent differences with different data, base models, and condition types. From the perspective of results, under the settings in our paper (with a larger batch size), running about 3k steps generally leads to generated results that are constrained by the conditional images.

**Comment by serend1p1ty** [comment]
@jiangzeyinzi
After training 5w steps, the model seems converge.
![image](https://github.com/modelscope/scepter/assets/39628662/fd436f35-3166-4914-ba8b-d9bae4b03324)
![image](https://github.com/modelscope/scepter/assets/39628662/32d6eb12-79cd-4fa3-ad09-4001278138c8)
![image](https://github.com/modelscope/scepter/assets/39628662/e56ca568-f87f-4500-98d8-dba5ffde4ebd)
For humans, generation seems to be not very stable.
I am a beginner in the field of text-to-image generation. ***Does the current result meet expectations***?

Next, I plan to train SCEdit on the larger LAION dataset. Your paper mentions that you used a fixed learning rate `5e-5`.
In order to reproduce the results in the paper, ***should I set the learning rate to `0.000125` if batch_size is `256`***? Because `0.000125*256/640=5e-5`.

Looking forward to your reply.

**Comment by jiangzeyinzi** [comment]
I believe it meets the expectations, as the COCO dataset contains a large number of images with multiple subjects and small faces, which are significant challenges in generation tasks. Training with premium data is a good approach. Additionally, a reasonable learning rate in experiments will not cause a particularly large effect deviation. It is recommended to set it at 5e-5 without having to follow the batch learning rate setting. I hope you achieve good results.

**Comment by serend1p1ty** [comment]
Thanks for you reply.
